Lib/argparse.py

```diff
@@ -2723,15 +2723,13 @@ def format_help(self):
         return formatter.format_help()
 
     def _get_formatter(self):
-        if isinstance(self.formatter_class, type) and issubclass(
-            self.formatter_class, HelpFormatter
-        ):
+        try:
             return self.formatter_class(
                 prog=self.prog,
                 prefix_chars=self.prefix_chars,
                 color=self.color,
             )
-        else:
+        except TypeError:
```

Comment:
TypeError seems a bit generic (you're calling user code).

What do you think about this?

`len([v.kind for (k, v) in inspect.signature(self.formatter_class).parameters.items() if k in ('y', 'z') and v.kind in (inspect._ParameterKind.POSITIONAL_OR_KEYWORD, inspect._ParameterKind.KEYWORD_ONLY)]) == 2
`

Reply:
That doesn't work: `FAILED (failures=2, errors=156)`

Reply:
Wait, it does, obviously need to replace `y` and `z` :)